Makefile

```diff
@@ -12,138 +12,258 @@
+	cd $(TOOLS_MOD_DIR) && \
+		go build -tags=tools -o $(TOOLS_BIN_DIR)/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
 
-.PHONY: e2e-bootstrap
-e2e-bootstrap: install-helm
-	apt-get update && apt-get install bats && apt-get install gettext-base -y
-	# Download and install Vault
-	vault -v | grep -q v$(VAULT_VERSION) || (curl -LO https://releases.hashicorp.com/vault/$(VAULT_VERSION)/vault_$(VAULT_VERSION)_linux_amd64.zip && unzip vault_$(VAULT_VERSION)_linux_amd64.zip && chmod +x vault && mv vault /usr/local/bin/)
-	# Download and install Azure CLI
+$(GOLANGCI_LINT): ## Build golangci-lint from tools folder.
+	cd $(TOOLS_MOD_DIR) && \
+		go build -tags=tools -o $(TOOLS_BIN_DIR)/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
+
+$(KUSTOMIZE): ## Build kustomize from tools folder.
+	cd $(TOOLS_MOD_DIR) && \
+		go build -tags=tools -o $(TOOLS_BIN_DIR)/kustomize sigs.k8s.io/kustomize/kustomize/v3
+
+## --------------------------------------
+## Testing Binaries
+## --------------------------------------
+
+$(HELM): ## Install helm3 if not present
+	helm version --short | grep -q v3 || (curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash)
+
+$(KIND): ## Download and install kind
+	kind --version | grep -q $(KIND_VERSION) || (curl -L https://github.com/kubernetes-sigs/kind/releases/download/v$(KIND_VERSION)/kind-linux-amd64 --output kind && chmod +x kind && mv kind /usr/local/bin/)
+
+$(AZURE_CLI): ## Download and install azure cli
 	curl -sL https://aka.ms/InstallAzureCLIDeb | bash
-ifndef TEST_WINDOWS
+
+$(BATS): ## Install bats for running the tests
+	bats --version | grep -q $(BATS_VERSION) || (apt-get -o Acquire::Retries=30 update && apt-get -o Acquire::Retries=30 install -y bats)
+
+$(KUBECTL): ## Install kubectl
 	curl -LO https://storage.googleapis.com/kubernetes-release/release/v$(KUBERNETES_VERSION)/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv kubectl /usr/local/bin/
-	make setup-kind
+
+$(TRIVY): ## Install trivy for image vulnerability scan
+	trivy -v | grep -q $(TRIVY_VERSION) || (curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v$(TRIVY_VERSION))
+
+$(BATS): ## Install bats for running the tests
+	bats --version | grep -q $(BATS_VERSION) || (apt-get -o Acquire::Retries=30 update && apt-get -o Acquire::Retries=30 install -y bats)
+
+$(ENVSUBST): ## Install envsubst for running the tests
+	envsubst -V || (apt-get -o Acquire::Retries=30 update && apt-get -o Acquire::Retries=30 install gettext-base -y)
+
+## --------------------------------------
+## Linting
+## --------------------------------------
+.PHONY: test-style
+test-style: lint lint-charts
+
+.PHONY: lint
+lint: $(GOLANGCI_LINT)
+	# Setting timeout to 5m as default is 1m
+	$(GOLANGCI_LINT) run --timeout=5m -v
+
+lint-full: $(GOLANGCI_LINT)
+	$(GOLANGCI_LINT) run -v --fast=false
```

Comment:
isn't `fast=false` the default? i think this is same as above.

Reply:
By default we set it to `true` (https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/441/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R23).